### DISCUSSION TITLE
How to use TOTG for time parameterization ?

### DISCUSSION THREAD

**Discussion by vegfood** [tesseract-robotics/tesseract_planning#237] [Q&A] [answered]
I follow the basic_cartesian_example in tesseract_ros_examples and i notice that it uses spline for time parameterization. In moveit, i can choose TOTG for time paramererization, so i wanna know how can this be done in tesseract ?

**Comment by Levi-Armstrong** [comment, marked as answer]
Yes, take a look at this `planning_server.loadDefaultProcessPlanners();` in the example. This is where all of the motion pipelines get added. You would need to copy the function for the pipelinei you are interested in and switch it out ISP with TOTG.

That example is using the trajopt pipeline so you would copy this function and replace it with TOTG where ISP is. 
https://github.com/tesseract-robotics/tesseract_planning/blob/f966455d0a75eb942aa45d69cabd8155706c4364/tesseract_process_managers/src/core/default_process_planners.cpp#L55-L120

Then register this function with the planning server under some name.

``` c++
planning_server.registerProcessPlanner(process_planner_names::TRAJOPT_PLANNER_NAME + "_TOTG", createTrajOptTOTGGenerator());
```
Then update this line `request.name = process_planner_names::TRAJOPT_PLANNER_NAME;` with `request.name = process_planner_names::TRAJOPT_PLANNER_NAME + "_TOTG";`

**Reply by vegfood** [reply]
Thanks, it works
